FinSearchUnified/Subscriber/Widgets.php

```diff
@@ -16,14 +40,73 @@ class Widgets implements SubscriberInterface
     public static function getSubscribedEvents()
     {
         return [
-            'Shopware_Controllers_Widgets_Listing::listingCountAction::before' => 'beforeListingCountAction'
+            'Shopware_Controllers_Widgets_Listing::listingCountAction::before' => 'beforeListingCountAction',
+            'Enlight_Controller_Action_PreDispatch_Widgets' => 'onWidgetsPreDispatch'
         ];
     }
 
+    /**
+     * @param Enlight_Event_EventArgs $args
+     *
+     * @throws Zend_Cache_Exception
+     * @throws Exception
+     */
+    public function onWidgetsPreDispatch(Enlight_Event_EventArgs $args)
+    {
+        /** @var Enlight_Controller_Request_RequestHttp $request */
+        $request = $args->get('request');
+        $url = $this->parseReferUrl($request);
+
+        if (strpos($url, 'search') !== false) {
+            Shopware()->Session()->isSearchPage = true;
+            Shopware()->Session()->isCategoryPage = false;
+        } else {
```

Comment:
Consider splitting up this huge else block into multiple smaller functions. This way devs who read this code the first time can see semantics by only reading the method names instead of checking the entire function just to know what a specific code part actually does.